datapackage_pipelines_budgetkey/common/object_storage.py

```diff
@@ -0,0 +1,114 @@
+import csv, os, io, contextlib
+from botocore.client import Config
+from botocore.exceptions import ClientError
+import boto3
+import json
+import datetime
+from contextlib import contextmanager
+from tempfile import mkdtemp
+
+
+def get_s3():
+    config = Config(signature_version=os.environ["S3_SIGNATURE_VERSION"]) if os.environ.get("S3_SIGNATURE_VERSION") else None
+    s3 = False
+    if os.environ.get("AWS_ACCESS_KEY_ID") and os.environ.get("AWS_SECRET_ACCESS_KEY"):
+        s3 = boto3.client('s3',
+                          endpoint_url=os.environ.get("S3_ENDPOINT_URL"),
+                          aws_access_key_id=os.environ["AWS_ACCESS_KEY_ID"],
+                          aws_secret_access_key=os.environ["AWS_SECRET_ACCESS_KEY"],
+                          config=config,
+                          region_name=os.environ.get("S3_REGION"))
+    return s3
+
+
+def exists(s3, bucket, object_name, min_size=None):
+    try:
+        res = s3.head_object(Bucket=bucket, Key=object_name)
+    except Exception:
+        res = False
+    if res and min_size:
+        return res['ContentLength'] > min_size
+    else:
+        return bool(res)
+
+
+def get_write_object_data(data):
+    if isinstance(data, str):
+        data = data.encode()
+    return io.BytesIO(data)
+
+
+def write(s3, bucket, object_name, data=None, file_name=None, create_bucket=True, public_bucket=False):
+    try:
+        if public_bucket:
+            s3.put_bucket_policy(Bucket=bucket, Policy=json.dumps({
```

Comment:
Why do we need to set bucket policy for each file?

Reply:
no need, good catch.. I'll fix it